swaggerpy/client.py

```diff
@@ -1,94 +1,131 @@
+               |
+               |  has many
+               |
+        +------v--------------+
+        |                     |
+        |     Resource        +------------------+
+        |                     |                  |
+        +------+--------------+         has many |
+               |                                 |
+               |  has many                       |
+               |                                 |
+        +------v--------------+           +------v--------------+
+        |                     |           |                     |
+        |     Operation       |           |    SwaggerModel     |
+        |                     |           |                     |
+        +------+--------------+           +---------------------+
+               |
+               |  uses
+               |
+        +------v--------------+
+        |                     |
+        |     HttpClient      |
+        |                     |
+        +---------------------+
+
+
+To get a client with caching
+
+.. code-block:: python
+
+        client = swaggerpy.client.get_client(api_docs_url)
+
+without caching
+
+.. code-block:: python
+
+        client = swaggerpy.client.SwaggerClient.from_url(api_docs_url)
 
-import json
+"""
+
+from swaggerpy.compat import json
 import logging
 import os.path
 import time
 import urllib
-from collections import namedtuple
-from datetime import datetime
 from urlparse import urlparse
 
 from yelp_uri import urllib_utf8
 
 import swagger_type
 from swaggerpy.http_client import APP_JSON, SynchronousHttpClient
-from swaggerpy.processors import SwaggerProcessor
-from swaggerpy.response import HTTPFuture, SwaggerResponse
-from swaggerpy.swagger_model import create_model_type, Loader
+from swaggerpy.response import HTTPFuture, post_receive
+from swaggerpy.swagger_model import (
+    Loader,
+    create_model_type,
+    is_file_scheme_uri,
+)
 from swaggerpy.swagger_type import SwaggerTypeCheck
 
 log = logging.getLogger(__name__)
 
-SWAGGER_SPEC_TIMEOUT_S = 300
+SWAGGER_SPEC_CACHE_TTL = 300
 
 
-class CachedClient(object):
-    """A wrapper to client which stores the last updated timestamp and the
-    timeout in secs. when the client expires
+class CacheEntry(object):
+    """An entry in the cache. Each item has it's own ttl.
 
-    :param swagger_client: Core SwaggerClient instance
-    :type swagger_client: :class:`SwaggerClient`
-    :param timeout: timeout in seconds after which the client expires
-    :type timeout: int
+    :param item: the item to cache
+    :param ttl: time-to-live in seconds after which the client expires
+    :type  ttl: int
     """
 
-    def __init__(self, swagger_client, timeout, timestamp=None):
-        self.swagger_client = swagger_client
-        self.timeout = timeout
+    def __init__(self, item, ttl, timestamp=None):
+        self.item = item
+        self.ttl = ttl
         self.timestamp = timestamp or time.time()
 
```

Comment:
Yup, I think we could be patching `time.time()` here. Going to leave for another branch though since it's not modified in this one.